src/company_cmd.cpp

```diff
@@ -875,6 +876,9 @@ CommandCost CmdCompanyCtrl(TileIndex tile, DoCommandFlag flags, uint32 p1, uint3
 			CompanyRemoveReason reason = (CompanyRemoveReason)GB(p1, 24, 8);
 			if (reason >= CRR_END) return CMD_ERROR;
 
+			/* We can't delete the last existing company in offline mode. */
+			if (!_networking && Company::GetNumItems() == 1) return CMD_ERROR;
```

Comment:
Maybe a good idea to throw a STR_ERROR so the user knows why this is failing?

Reply:
In theory, with second commit, the call should not happen.
And for an error message to be displayed, it seems all `DoCommand()` calls for that command need to be updated with `CMD_MSG()`

Reply:
Someone wise told me a few weeks ago:

`CMD_MSG` is the default error shown when there is no other after a `CMD_ERROR`.
Use `return_cmd_error(STR_ERROR_XXX);` is you want to return a specific one :)

Anyway, if you covered all routes via UI elements, I am fine with that. It just didn't read like that, hence the comment :D

Reply:
But `return_cmd_error(STR_ERROR_XXX)` has no visible effect if there was no `CMD_MSG`